src/dotnet-test-xunit/Program.cs

```diff
@@ -181,34 +182,32 @@ static void OnUnhandledException(object sender, UnhandledExceptionEventArgs e)
         static List<IRunnerReporter> GetAvailableRunnerReporters()
         {
             var result = new List<IRunnerReporter>();
-            var runnerPath = Path.GetDirectoryName(AppContext.BaseDirectory);
+            var dependencyModel = DependencyContext.Load(typeof(Program).GetTypeInfo().Assembly);
+            var assemblyNames = new List<RuntimeAssembly>(dependencyModel.RuntimeLibraries.SelectMany(r => r.Assemblies));
 
-            foreach (var dllFile in Directory.GetFiles(runnerPath, "*.dll").Select(f => Path.Combine(runnerPath, f)))
+            foreach (var assemblyName in assemblyNames)
             {
-                TypeInfo[] types;
-
-                var dllName = Path.GetFileNameWithoutExtension(dllFile);
-
+                Assembly assembly;
                 try
                 {
-                    var assembly = Assembly.Load(new AssemblyName(dllName));
-                    types = assembly.DefinedTypes.ToArray();
+                    assembly = Assembly.Load(assemblyName.Name);
+
                 }
                 catch
                 {
                     continue;
                 }
 
-                foreach (var type in types)
+                foreach (var type in assembly.DefinedTypes)
                 {
-                    if (type == null || type.IsAbstract || type == typeof(DefaultRunnerReporter).GetTypeInfo() || type.ImplementedInterfaces.All(i => i != typeof (IRunnerReporter)))
+                    if (type == null || type.IsAbstract || type == typeof(DefaultRunnerReporter).GetTypeInfo() || type.ImplementedInterfaces.All(i => i != typeof(IRunnerReporter)))
```

Comment:
Ehh, this is a port of external code which would need to be merged someday. I'd rather not change it.